### PR TITLE
Show media details fragment when on clicking image of  Peer Review

### LINKED ISSUE #2763: Peer review: show description
Description often says why a picture is special. For instance:

Title: Abdel Rawid
Description: Soccer player in the 2020 Iraqi Olympics team

That can change the reviewer's perception from "_Hum probably another worthless selfie_" to "_Whoa this is a very valuable picture_".

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**

Fixes #2763 

What changes did you make and why?

 * Currently on clicking the image of peer review it does nothing. Now it will show the media detail fragment on clicking the image of peer review 
* Add javadoc
* Reformated code of  ReviewActivity.xml

**Tests performed (required)**

Tested {Prodebug,} on {Redmi note 4,redmi y3} with API level {29,25}

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul please review this one

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -332,8 +332,13 @@ public void onResume() {
                 Label.valuesAsList()), categoryRecyclerView, categoryClient, this);
         categoryRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
         categoryRecyclerView.setAdapter(categoryEditSearchRecyclerViewAdapter);
+        // detail proovider is null when fragment is show in review activity
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:335]: proovider -> provider
show -> shown

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by madhurgupta10** [comment]
@Prince-kushwaha can you add unit tests for the all-new methods?

**Comment by nicolas-raoul** [comment]
It works great :-)

**Comment by Prince-kushwaha** [comment]
@madhurgupta10 yes i add unit test for methods

**Comment by Prince-kushwaha** [comment]
@madhurgupta10  help me how I run Unit Test-ReviewActivityTest It does not have option to run the test

**Comment by madhurgupta10** [comment]
You can run tests from Android Studio like this

[image: Capture.PNG]

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 2]
```diff
@@ -258,7 +258,7 @@ public ReviewImageFragment getInstanceOfReviewImageFragment(){
     /**
      * set up the media detail fragment when click on the review image
      */
-    private void setUpMediaDetailFragment() {
+    public void setUpMediaDetailFragment() {
```
Comment (madhurgupta10) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:261]: No need to make them public for unit tests

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ public void onBackPressed() {
     /**
      * set up media detail fragment after orientation change
      */
-    private void setUpMediaDetailOnOrientation() {
+    public void setUpMediaDetailOnOrientation() {
```
Comment (madhurgupta10) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:291]: No need to make them public for unit tests

File: app/src/test/kotlin/fr/free/nrw/commons/review/ReviewActivityTest.kt [hunk 4]
```diff
@@ -101,4 +101,22 @@ class ReviewActivityTest {
         activity.onOptionsItemSelected(menuItem)
     }
 
+    @Test
+    @Throws(Exception::class)
+    fun testSetUpMediaDetailFragment() {
+        activity.setUpMediaDetailFragment()
```
Comment (madhurgupta10) [inline app/src/test/kotlin/fr/free/nrw/commons/review/ReviewActivityTest.kt:107]: For private methods test like this

```
val method: Method = ClassToTest::class.java.getDeclaredMethod(
            "methodToTest"
        )
        method.isAccessible = true
        method.invoke(classTestObject)
```

**Review by madhurgupta10 — CHANGES_REQUESTED** [review]

**Comment by madhurgupta10** [comment]
@Prince-kushwaha can you also post the screenshot of the code coverage for this file?

**Comment by Prince-kushwaha** [comment]
@madhurgupta10 code coverage for this file
 
tes

**Review by nicolas-raoul — APPROVED** [review]
Looks OK from my point of view.
@madhurgupta10 anything to add? :-)

**Comment by madhurgupta10** [comment]
@Prince-kushwaha I tested your PR, your tests are not covering most of the new lines added, if you can cover maximum lines then it would be great (Can be done in a separate PR)

@nicolas-raoul I think we can merge this one!

Capture

**Review by madhurgupta10 — APPROVED** [review]

**Review by neslihanturan — APPROVED** [review]
Works as intended, nice job @Prince-kushwaha , thanks!

**Pull request merged**